ui/apps/platform/src/Components/ColumnManagementButton.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+
+import { ColumnManagementModal } from '@patternfly/react-component-groups';
+import { Button, ButtonVariant } from '@patternfly/react-core';
+
+import { ColumnConfig, ManagedColumns } from 'hooks/useManagedColumns';
+
+export type ColumnManagementButtonProps<ColumnKey extends string> = {
+    managedColumnState: ManagedColumns<ColumnKey>;
+};
+
+function ColumnManagementButton<ColumnKey extends string>({
+    managedColumnState,
+}: ColumnManagementButtonProps<ColumnKey>) {
+    const [isOpen, setOpen] = useState(false);
+    const { columns, setColumns } = managedColumnState;
+    const enabledColumnCount = Object.values<ColumnConfig>(columns).filter(
+        ({ isShown }) => isShown
+    ).length;
+
+    return (
+        <>
+            <ColumnManagementModal
+                appliedColumns={Object.values(columns)}
+                applyColumns={(newColumns) => {
+                    const nextState = Object.fromEntries(
+                        newColumns.map(({ key, isShown }) => [key, isShown ?? false])
+                    );
+                    setColumns(nextState);
+                }}
+                isOpen={isOpen}
```

Comment:
What are pro and con:
* Unconditional rendering of instance with `isOpen` prop.
* Conditional rendering like `{isOpen && …}` with `isOpen={true}` prop

Reply:
In this case, I don't think there is any pro to conditional rendering. An `isOpen` prop of `false` will cause the PatternFly component to be rendered as `null`, and the conditional rendering on our end would only increase the JSX indent and add a couple of lines of code.